File: src/main/java/io/shodo/BasketInformations.java
package src.main.java.io.shodo;

import java.util.HashMap;

/**
 * Created by thomas on 02/12/2019.
 * This class is a basket
 */
public class BasketInformations {

    // The product of the basket
    static HashMap<String, Integer> map = new HashMap<String, Integer>();

    // The fact that the basket has promo code
    private static boolean codeDePromotion = false;

    public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
        if (isPromoCode) {
            codeDePromotion = true;
        } else {
            map.put(product, price);
        }
    }

    public Long getBasketPrice(boolean inCents) {
        Integer v = 0;
        for (String s : map.keySet()) {
            v += map.get(s);
        }
        if (codeDePromotion) {
            v -= 100;
        }
        return inCents ? v * 100 : Long.valueOf(v);
    }
}

File: typescript/src/world-company-remuneration/basket-informations.ts
const K = "K";

/**
 * Created by thomas on 02/12/2019.
 * This class is a basket
 */

export class BasketInformations {
    // The product of the basket
    static map: Map<string, number> = new Map<string, number>()

    addProductToBasket(product: string, price: number) {
        BasketInformations.map.set(product, price)
    }

    getBasketPrice(inCents: boolean): Number {
        var v = 0;
        for (let s of Array.from(BasketInformations.map.values())) {
            v += s;
        }
        return inCents ? new Number(v * 100) : Number(v)
    }

    resetBasket() {
        this.buyBasket();
    }

    buyBasket() {
        BasketInformations.map.clear();
    }

    isBasketContains(produit: string): boolean {
        var found: boolean = false;
        for (let s of Array.from(BasketInformations.map.keys())) {
            if (s == produit) found = true;
        }
        return found;
    }


    mixWithBasket(b: BasketInformations) {
        for (let [s,z] of Array.from(BasketInformations.map)) {
            BasketInformations.map.set(s,z)
        }
    }
}
